lib/cli/run.js

```diff
@@ -299,8 +299,14 @@ exports.builder = yargs =>
     .number(types.number)
     .alias(aliases);
 
-exports.handler = argv => {
+exports.handler = async function(argv) {
   debug('post-yargs config', argv);
   const mocha = new Mocha(argv);
-  runMocha(mocha, argv);
+
+  try {
+    await runMocha(mocha, argv);
+  } catch (err) {
+    console.error(err.stack || `Error: ${err.message || err}`);
+    process.exit(1);
+  }
```

Comment:
I'm unsure wether this will be sufficient to avoid "unhandled promise rejections". When a promise further down the call stack is failing and not catched.  We will see ....

Reply:
If we're dealing with all the promises down the call stack correctly, then there shouldn't be a problem.